src/file.rs

```diff
@@ -43,11 +77,66 @@ impl HasPath for PubkyAppFile {
 }
 
 impl Validatable for PubkyAppFile {
-    // TODO: content_type validation.
+    fn sanitize(self) -> Self {
+        let name = self.name.trim().chars().take(MAX_NAME_LENGTH).collect();
+
+        let sanitized_src = self
+            .src
+            .trim()
+            .chars()
+            .take(MAX_SRC_LENGTH)
+            .collect::<String>();
+
+        let src = match Url::parse(&sanitized_src) {
+            Ok(_) => Some(sanitized_src),
+            Err(_) => None, // Invalid src URL, set to None
+        };
+
+        let content_type = self.content_type.trim().to_string();
+
+        Self {
+            name,
+            created_at: self.created_at,
+            src: src.unwrap_or("".to_string()),
+            content_type,
+            size: self.size,
+        }
+    }
+
     fn validate(&self, id: &str) -> Result<(), String> {
         self.validate_id(id)?;
-        // TODO: content_type validation.
-        // TODO: size and other validation.
+
+        // Validate name
+        let name_length = self.name.chars().count();
+
+        if !(MIN_NAME_LENGTH..=MAX_NAME_LENGTH).contains(&name_length) {
+            return Err("Validation Error: Invalid name length".into());
+        }
+
+        // Validate src
+        if self.src.chars().count() == 0 {
+            return Err("Validation Error: Invalid src".into());
+        }
+        if self.src.chars().count() > MAX_SRC_LENGTH {
+            return Err("Validation Error: src exceeds maximum length".into());
+        }
+
+        // validate content type
+        match Mime::from_str(&self.content_type) {
+            Ok(mime) => {
+                if !VALID_MIME_TYPES.contains(&mime.essence_str()) {
```

Comment:
Declared mime type could be `pdf` and blob be a `exe` , right? On click on the browser from pubky-app, the user will download a `.exe` ?

Reply:
yes. The user doesn't really have an incentive to lie about this.  
Ultimately of course, we can have some security checker that runs the files through some malware detector, but mind you, something like this needs to be implemented on the homeserver first. The same goes for content type and sniffing the actual file type of the blob.

Reply:
> The user doesn't really have an incentive to lie about this.

Can you elaborate?

Reply:
I'm saying for normal users there's no need to lie about the content type of their files.  
For malicious intent, you don't have to lie about the content type. IMO, If there are to be checks for this, it should be on the homeserver, not here.